### PR TITLE
MISUV-672 - Updated to allow multiple self employment data storage

### PR DESCRIPTION
**Pull request by AlexRimmerHMRC**

### CODE REVIEW

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/check_your_answers.scala.html [hunk 1]
```diff
@@ -36,48 +38,51 @@
...
-@tradingStartDate = @{
+
+@tradingStartDate(businessStartDate: BusinessStartDate, businessId: String) = @{
     import implicitDateFormatter.LongDate
     val question: String = messages("check_your_answers.trading_start_date")
-    val answer: String = answers.businessStartDate.startDate.toLocalDate.toLongDate
-    val changeUrl: String = uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessStartDateController.show(id, isEditMode = true).url
-
+    val answer: String = businessStartDate.startDate.toLocalDate.toLongDate
+    val changeUrl: String = uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessStartDateController.show(businessId, isEditMode = true).url
```
Comment (chandraAndem) [inline app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/check_your_answers.scala.html:46]: As mentioned above we need to pass in index of the business which we intend to change.

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/check_your_answers.scala.html [hunk 2]
```diff
@@ -36,48 +38,51 @@
...
 }
 
-@businessName = @{
+@businessName(businessName: BusinessNameModel, businessId: String) = @{
     val question: String = messages("check_your_answers.business_name")
-    val answer: String = answers.businessName.businessName
-    val changeUrl: String = uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessNameController.show(id, isEditMode = true).url
-
+    val answer: String = businessName.businessName
+    val changeUrl: String = uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessNameController.show(businessId, isEditMode = true).url
```
Comment (chandraAndem) [inline app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/check_your_answers.scala.html:53]: same as above.

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/check_your_answers.scala.html [hunk 3]
```diff
@@ -36,48 +38,51 @@
...
 }
 
-@businessTrade = @{
+@businessTrade(businessTrade: BusinessTradeNameModel, businessId: String) = @{
     val question: String = messages("check_your_answers.business_trade")
-    val answer: String = answers.businessTradeName.businessTradeName
-    val changeUrl: String = uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessTradeNameController.show(id, isEditMode = true).url
-
+    val answer: String = businessTrade.businessTradeName
+    val changeUrl: String = uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessTradeNameController.show(businessId, isEditMode = true).url
```
Comment (chandraAndem) [inline app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/check_your_answers.scala.html:60]: same as above

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessStartDateController.scala [hunk 4]
```diff
@@ -23,43 +23,49 @@ import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Request}
...
     authService.authorised() {
-      incomeTaxSubscriptionConnector.getSelfEmployments[BusinessStartDate](BusinessStartDateController.businessStartDateKey).map {
-        case Right(businessStartDateData) =>
-          Ok(view(form.fill(businessStartDateData), id, isEditMode))
-        case error => throw new InternalServerException(error.toString)
+      multipleSelfEmploymentsService.fetchAllBusinesses.flatMap {
+        case Right(businesses) =>
+          multipleSelfEmploymentsService.fetchBusinessStartDate(id).map {
+            case Right(businessStartDateData) =>
+              println(businessStartDateData)
```
[the remark this answers is not in the document]
Comment (AlexRimmerHMRC) [reply app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessStartDateController.scala:64]: oops :man_facepalming:

**Pull request merged**